synapse/events/utils.py

```diff
@@ -475,14 +475,14 @@ def serialize_event(
     if config.as_client_event:
         d = config.event_format(d)
 
-    # If the event is a redaction, copy the redacts field from the content to
-    # top-level for backwards compatibility.
-    if (
-        e.type == EventTypes.Redaction
-        and e.room_version.updated_redaction_rules
-        and e.redacts is not None
-    ):
-        d["redacts"] = e.redacts
+    # If the event is a redaction, either copy the redacts field from the content to
+    # top-level for backwards compatibility OR from the redacts field to the content
+    # field for forwards compatibility.
+    if e.type == EventTypes.Redaction and e.redacts is not None:
+        if e.room_version.updated_redaction_rules:
+            d["redacts"] = e.redacts
+        else:
+            d["content"]["redacts"] = e.redacts
```

Comment:
My reading of https://github.com/matrix-org/matrix-spec/pull/1604/files:

In V11:
 -  the "raw" event stores the redacts field in content
  - Synapse sets `e.redacts` by parsing from the content dict
  - for the CSAPI, copy e.redacts into a top-level redacts field

In V10 and older:

 -  the "raw" event stores the redacts field in a top-level field
  - Synapse sets `e.redacts` by parsing that field
  - for the CSAPI, copy e.redacts into the content

In both situations, clients see the `redacts` field twice: at the top-level, and in content.